fundamentals/aot/diagnostics/Rdg13/Program.cs

```diff
@@ -0,0 +1,90 @@
+#define RDG13F //   RDG13 RDG13F
+#if NEVER
+#elif RDG13
+// <snippet_1>
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.TypeInfoResolverChain.Insert(0,
+                                       AppJsonSerializerContext.Default);
+});
+
+var app = builder.Build();
+
+app.MapGet("/v1/todos", 
+    ([FromKeyedServices("primary")][FromServices] ITodoService todoService)
+   => Results.Ok(todoService.GetTodos()));
+
+app.Run();
+
+record Todo(int Id, string Task);
+interface ITodoService
+{
+    Todo[] GetTodos();
+}
+
+[JsonSerializable(typeof(Todo[]))]
+internal partial class AppJsonSerializerContext : JsonSerializerContext
+{
+
+}
+// </snippet_1>
+#elif RDG13F
+// <snippet_1f>
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+
+builder.Services.AddScoped<ITodoService, TodoService>();
```

Comment:
@captainsafia I'm registering ITodoService here but at run time I get `System.InvalidOperationException:` 'No service for type 'ITodoService' has been registered.'

cc @timdeschryver

Reply:
Hi Rick, you can fix this by registering the service using `AddKeyedScoped`.

https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8#keyed-di-services

```cs
builder.Services.AddKeyedScoped<ITodoService, TodoService>("primary");
```